### PR TITLE
fix: fix ut flakiness

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes

Fix the unit test failure
https://github.com/Azure/fleet/actions/runs/9601760461/attempts/1

Fixes #

I have:

- [ ] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/controllers/workgenerator/controller_test.go [hunk 1]
```diff
@@ -677,7 +678,6 @@ func TestSetBindingStatus(t *testing.T) {
 					},
 				},
 			},
-			maxFailedResourcePlacementLimit: 2,
```
Comment (ryanzhang-oss) [inline pkg/controllers/workgenerator/controller_test.go:680]: looks like we don't test when it's 2 anymore?

File: pkg/controllers/workgenerator/controller_test.go [hunk 2]
```diff
@@ -971,10 +970,26 @@ func TestSetBindingStatus(t *testing.T) {
...
+			if tt.maxFailedResourcePlacementLimit != nil {
+				maxFailedResourcePlacementLimit = *tt.maxFailedResourcePlacementLimit
+			}
 			binding := &fleetv1beta1.ClusterResourceBinding{}
 			setBindingStatus(tt.works, binding)
 			got := binding.Status.FailedPlacements
+			// setBindingStatus is using map to populate the failedResourcePlacement.
+			// There is no default order in traversing the map.
+			// When the result of failedResourcePlacement exceeds the limit, the result will be truncated and cannot be
+			// guaranteed.
```
Comment (ryanzhang-oss) [inline pkg/controllers/workgenerator/controller_test.go:982]: I was aware of that but thought it was not super random. Now that we don't test the limit anymore, can we just remove the limit from the test?

File: pkg/controllers/workgenerator/controller_test.go (hunk 1 above)
Comment (zhiying-lin) [reply pkg/controllers/workgenerator/controller_test.go:680]: The default is 100. if the failed number < limit, 2 or 100 does not matter.

File: pkg/controllers/workgenerator/controller_test.go (hunk 2 above)
Comment (zhiying-lin) [reply pkg/controllers/workgenerator/controller_test.go:982]: we're testing limit in line 821

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**